MinimizingLateness/MinimizingLateness.java

```diff
@@ -22,39 +22,46 @@ public Schedule(int t, int d) {
   public static void main(String[] args) throws IOException {
     StringTokenizer token;
 
-    BufferedReader in = new BufferedReader(new FileReader("MinimizingLateness/lateness_data.txt"));
-    String ch = in.readLine();
-    if (ch == null || ch.isEmpty()) {
-      in.close();
-      return;
-    }
-    int indexCount = Integer.parseInt(ch);
-    System.out.println("Input Data : ");
-    System.out.println(indexCount); // number of operations
-    Schedule[] array = new Schedule[indexCount]; // Create an array to hold the operation
-    int i = 0;
-    while ((ch = in.readLine()) != null) {
-      token = new StringTokenizer(ch, " ");
-      // Include the time required for the operation to be performed in the array and the time it
-      // should be completed.
-      array[i] =
-          new Schedule(Integer.parseInt(token.nextToken()), Integer.parseInt(token.nextToken()));
-      i++;
-      System.out.println(array[i - 1].t + " " + array[i - 1].d);
-    }
+    BufferedReader in = null;
+    try{
+      new BufferedReader(new FileReader("MinimizingLateness/lateness_data.txt"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *BufferedReader*. Currently, there are execution paths that do not contain closure statements, for example, when an else condition at line 29 evaluates to true. Ensure the object returned by *BufferedReader()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).